File: Leetcode/560-Subarray Sum Equals K.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int ans = 0;
        int tmp = 0;
        for(int i=0 ; i<nums.length ;i++){
            tmp=0;
            for(int j=i ; j<nums.length ; j++){
                tmp+=nums[j];
                if(tmp==k){
                    ans++;
                }
            }
        }
        return ans;
    }
}

File: Leetcode-Solutions/560-Subarray_Sum_Equals_K.cpp
#include <iostream>
using namespace std;
class Solution {
public:
  int subarraySum(vector<int> &nums, int k) {
    unordered_map<int, int> mp = {{0, 1}};
    int ans = 0;
    int tot = 0;
    for (int i = 0; i < nums.size(); i++) {
      tot += nums[i];
      int tmp = tot - k;
      ans += mp[tmp];
      mp[tot]++;
    }
    return ans;
  }
};
